janusgraph-core/src/main/java/org/janusgraph/core/JanusGraph.java

```diff
@@ -57,6 +57,11 @@
         reason = "JanusGraph test graph computer (FulgoraGraphComputer) " +
             "currently does not support graph filters but does not throw proper exception because doing so breaks numerous " +
             "tests in gremlin-test ProcessComputerSuite.")
+@Graph.OptOut(
```

Comment:
New behavior may fail users code/tests, who reuse transaction listeners within auto tx, so this is breaking change, may be add some notes to upgrade/release notes?

Reply:
That is a good point. Now I am wondering if we should introduce this breaking change, or leave this memory leak as it is (provided there is no other way to fix the leak caused by transactionListeners).

Alternatively, we can let users know that after a transaction is done, they will have to call `graph.tx().addTransactionListener(listener);` explicitly if they want to reuse the listener in the new transaction.

Reply:
@mad Do you want to review again?